sqlglot/dialects/mysql.py

```diff
@@ -617,6 +613,37 @@ def _parse_chr(self) -> t.Optional[exp.Expression]:
 
             return self.expression(exp.Chr, **kwargs)
 
+        def _parse_group_concat(self) -> t.Optional[exp.Expression]:
+            def concat_exprs(
+                node: t.Optional[exp.Expression], exprs: t.List[exp.Expression]
+            ) -> exp.Expression:
+                if isinstance(node, exp.Distinct) and len(node.expressions) > 1:
+                    concat_exprs = [self.expression(exp.Concat, expressions=node.expressions)]
```

Comment:
We need to set `safe=True` in this `Concat` instantiation and the one below, because MySQL allows `CONCAT(1, 2)` but other dialects don't. With `safe` we'll [cast](https://github.com/tobymao/sqlglot/blob/11fe3692090dc0feb940994ba6bfe839b4fdfce1/sqlglot/generator.py#L2469) the args to strings when transpiling.

Reply:
Let's add a test for this with the fix.